pkg/game/test/game_test.go

```diff
@@ -0,0 +1,90 @@
+
+func TestMakeTurn(t *testing.T) {
+	minitileSet := tilesets.OrderedMiniTileSet1()
+	deckStack := stack.NewOrdered(minitileSet.Tiles)
+	deck := deck.Deck{Stack: &deckStack, StartingTile: minitileSet.StartingTile}
+	game, err := game.NewFromDeck(deck, nil, 2)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	test.MakeTurn(game, t, position.New(0, 1), test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Left, FeatureType: feature.Road})
+
+	board := game.GetBoard()
+	ptile, ok := board.GetTileAt(position.New(0, 1))
+	if !ok {
+		t.Fatal("expected to find a tile at (0, 1)")
+	}
+
+	tile := elements.ToTile(ptile)
+	expected := minitileSet.Tiles[0]
+	if !tile.Equals(expected) {
+		t.Fatalf("expected %#v, got %#v instead", expected, tile)
+	}
+
+	feat := ptile.GetPlacedFeatureAtSide(side.Left, feature.Road)
+	if feat.Meeple.Type != elements.NormalMeeple {
+		t.Fatalf("expected normal meeple on road tile feature, got %#v instead", feat.Meeple.Type)
+	}
+}
+
+func TestMakeTurnValidCheck(t *testing.T) {
+	// create game
+	minitileSet := tilesets.OrderedMiniTileSet2()
+	deckStack := stack.NewOrdered(minitileSet.Tiles)
+	deck := deck.Deck{Stack: &deckStack, StartingTile: minitileSet.StartingTile}
+	game, err := game.NewFromDeck(deck, nil, 4)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	test.MakeTurnValidCheck(game, t, position.New(0, 1), test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Bottom, FeatureType: feature.Road}, false, 1) // do any wrong move, and catch it
+	test.MakeTurnValidCheck(game, t, position.New(1, 0), test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Bottom, FeatureType: feature.Road}, true, 1)  // do any correct move
+}
+
+func TestCheckMeeplesAndScore(t *testing.T) {
+	// create game
+	minitileSet := tilesets.OrderedMiniTileSet2()
+	deckStack := stack.NewOrdered(minitileSet.Tiles)
+	deck := deck.Deck{Stack: &deckStack, StartingTile: minitileSet.StartingTile}
+	game, err := game.NewFromDeck(deck, nil, 4)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	test.CheckMeeplesAndScore(game, t, []uint32{0, 0}, []uint8{7, 7}, 1)
+}
+
+func TestVerifyMeepleExistence(t *testing.T) {
+	// create game
+	minitileSet := tilesets.OrderedMiniTileSet2()
+	deckStack := stack.NewOrdered(minitileSet.Tiles)
+	deck := deck.Deck{Stack: &deckStack, StartingTile: minitileSet.StartingTile}
+	game, err := game.NewFromDeck(deck, nil, 4)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	pos := position.New(1, 0)
+	test.MakeTurnValidCheck(game, t, pos, test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Bottom, FeatureType: feature.Road}, true, 1) // do any correct move
+
+	test.VerifyMeepleExistence(t, game, pos, side.Bottom, feature.Road, true, 1)           // verify that meeple truly exists
+	test.VerifyMeepleExistence(t, game, pos, side.BottomLeftEdge, feature.Field, false, 1) // verify that meeple truly exists
+}
```

Comment:
The only thing that these tests seem to do is check that the functions don't outright panic. Could you check in some way that they actually do what they should?

Reply:
addedd check if tile and meeple was placed to TestMakeTurnValidCheck, however idk how am i supposed to fix other tests.  Those functions do not return anything, so the only thing i can check is that they don't fail when not expected.

Reply:
I'd suggest adding test cases for when they do fail. I am guessing the difficulty here is that `t.Fatal` will automatically mark the whole test as failing - you can change functions that accept `*testing.T` to instead accept an interface such as this:
```
type T interface {
	Fatal(args ...any)
	Fatalf(format string, args ...any)
}
```
This interface will be satisfied by `*testing.T`.

Then, in tests, you can define your own type that satisfies this interface and instead of failing panics which allows you to catch it.

Reply:
the interface is made, for catching fails!